modules/openapi-generator-cli/src/main/java/org/openapitools/codegen/cmd/Generate.java

```diff
@@ -240,6 +240,9 @@
     @Option(name = {"--generate-alias-as-model"}, title = "generate alias (array, map) as model", description = CodegenConstants.GENERATE_ALIAS_AS_MODEL_DESC)
     private Boolean generateAliasAsModel;
 
+    @Option(name = {"--discriminator-explicit-mapping-verbose"}, title = "explicitly define a discriminator on all models which have one, add verbose mapping to that discriminator, validate discriminators", description = CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE_DESC)
```

Comment:
This option needs clarification since it's currently doing a lot.

1. Can we remove the "verbose" since that's not really relevant to the user?
2. The term `mapping` would indicate to users who are familiar with our options that this would take a string in the format `key=value`, which is not the case.
3. The option is triggering OpenAPI 3.x logic, but doesn't indicate that clearly here.

Might I recommend changing this to something more like:

```
@Option(name = {"--legacy-discriminators"}, title = "Support legacy logic for evaluating discriminators", description = CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE_DESC)
private Boolean legacyDiscriminators;
```

This value would need to default to `true` until the new logic becomes default (I don't recall if it's now default or not).

The value would then have to be applied to CodegenConfigurator, which is not done in `Generate.java` here. The same default would have to be done in the Maven and Gradle plugins, with a task to add the feature option to those. We also have a new SBT plugin (pending first release) where this would have to be added.

Reply:
> This option needs clarification since it's currently doing a lot.
> 
> 1. Can we remove the "verbose" since that's not really relevant to the user?
> 2. The term `mapping` would indicate to users who are familiar with our options that this would take a string in the format `key=value`, which is not the case.
> 3. The option is triggering OpenAPI 3.x logic, but doesn't indicate that clearly here.
> 
> Might I recommend changing this to something more like:
> 
> ```
> @Option(name = {"--legacy-discriminators"}, title = "Support legacy logic for evaluating discriminators", description = CodegenConstants.DISCRIMINATOR_EXPLICIT_MAPPING_VERBOSE_DESC)
> private Boolean legacyDiscriminators;
> ```
> 
> This value would need to default to `true` until the new logic becomes default (I don't recall if it's now default or not).
> 
> The value would then have to be applied to CodegenConfigurator, which is not done in `Generate.java` here. The same default would have to be done in the Maven and Gradle plugins, with a task to add the feature option to those. We also have a new SBT plugin (pending first release) where this would have to be added.

I think the name should be even more explicit. Without this PR, the existing OAS v3 is buggy. It's not just that it would be legacy, it is not fully compliant with the OAS oneOf/allOf/anyOf spec. At least that's my assessment for python, python-experimental, go, go-experimental, powershell and all java clients. Until a language generator enables this parameter, it should not claim support for anyOf/allOf/oneOf, or maybe it should only claim partial support.

Reply:
@sebastien-rosset allowing the user to continue using the same previous behavior (legacy), means we don't break people who have written customization to overcome any limitations.

Maybe "legacy discriminator behavior"? As it is now, I don't think we could meet this with such a confusing option.

Consider for instance someone with a 2.0 spec shouldn't assume that setting this value to true gets them 3.0 behavior. As it is now, that's what it sounds like. A 3.0 spec shouldn't change generated output drastically. I agree this makes the logic more correct, but again people are about to extend generators as well as templates. Any user who has specified vendor extensions or written a custom generator would be blocked from updates because of this change, and such a user inspections options wouldn't understand this is the option to change.

Reply:
> @sebastien-rosset allowing the user to continue using the same previous behavior (legacy), means we don't break people who have written customization to overcome any limitations.
> 
> Maybe "legacy discriminator behavior"? As it is now, I don't think we could meet this with such a confusing option.

I agree the current option name could be improved.

> 
> Consider for instance someone with a 2.0 spec shouldn't assume that setting this value to true gets them 3.0 behavior. As it is now, that's what it sounds like. A 3.0 spec shouldn't change generated output drastically. I agree this makes the logic more correct, but again people are about to extend generators as well as templates. Any user who has specified vendor extensions or written a custom generator would be blocked from updates because of this change, and such a user inspections options wouldn't understand this is the option to change.

Sure, but don't you think at some point all language generators should enable the new behavior by default (as implemented in this PR)? The tool should support all OAS 3.x features by default. Otherwise, somebody who is downloading OpenAPITools for the first time and compiles a OAS v3 spec that contains oneOf/allOf/anyOf.

If there needs to be a differentiation between 2.x and 3.x maybe that could be done selectively based on the version?

Reply:
Yes, but unfortunately you can't really have non-breaking behavior for 3.0 (bug fix, spec support) and potentially break the behavior for 2.0 at the same time. We have to play it safer, especially since feature requests indicate more people are on 2.0 specs still, and we don't have telemetry to know differently. If we can do these simultaneously, then there's no need to a user-facing option.

Reply:
This has been changed to legacyDiscriminatorBehavior